src/figuresClasses.ts

```diff
@@ -1,11 +1,83 @@
-export interface Figure {}
+type Shape = `triangle` | `circle` | `rectangle`;
+type Color = `red` | `green` | `blue`;
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+    public shape: Shape = `triangle`,
+  ) {
+    const sorted = [a, b, c].sort((num1, num2) => num2 - num1);
+
+    if (this.a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('your error message');
+    }
+
+    if (sorted[0] >= sorted[1] + sorted[2]) {
+      throw new Error('your error message');
+    }
+  }
+
+  getArea(): number {
+    const semiperemeter = (this.a + this.b + this.c) * 0.5;
+    const area = Math.sqrt(
+      semiperemeter *
+        (semiperemeter - this.a) *
+        (semiperemeter - this.b) *
+        (semiperemeter - this.c),
+    );
+
+    return +area.toFixed(2);
+  }
+}
 
-export class Circle implements Figure {}
+export class Circle implements Figure {
+  constructor(
+    public color: Color,
+    public radius: number,
+    public shape: Shape = `circle`,
+  ) {
```

Comment:
[CODE KNOWLEDGE] - When you declare a constructor parameter with an access modifier TypeScript implicitly creates a property on the class with the same name as the parameter and assigns the parameter value to that property. You don't need to declare them again inside the constructor body. You can remove the `shape` initialization since it's already being handled by TypeScript.